react/components/LoginContent.js

```diff
@@ -161,7 +161,7 @@ class LoginContent extends Component {
 
   state = {
     sessionProfile: this.props.profile,
-    isOnInitialScreen: !this.props.profile,
+    isOnInitialScreen: !this.props.profile || !this.props.profile.isAuthenticated,
```

Comment:
Maybe like this would be more readable

```suggestion
    isOnInitialScreen: !(this.props.profile && this.props.profile.isAuthenticated),
```